src/main/kotlin/de/bund/digitalservice/useid/service/IdentityService.kt

```diff
@@ -0,0 +1,27 @@
+package de.bund.digitalservice.useid.service
+
+import de.bund.digitalservice.useid.datasource.SessionDataSource
+import de.bund.digitalservice.useid.model.IdentityAttributes
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+
+@Service
+class IdentityService(private val sessionDataSource: SessionDataSource) {
+    fun getIdentity(sessionId: String): Mono<IdentityAttributes> {
+        return Mono.just(IdentityAttributes("firstname", "lastname"))
+            .filter {
```

Comment:
Why is this a `filter`? I think I don't understand, how this works.

Reply:
so in the first step client request `refreshAddress` and `requestAttributes` which backend will response with `sessionId`, this sessionId is stored in-memory in the source, so that later when client request the identity data with a sessionId (I think step 17 in the diagram), backend checks if the `sessionId` is valid `sessionId` from the first exchange.